swift/Multiplay/Multiplay.swift

```diff
@@ -0,0 +1,323 @@
+            json = try? JSON(data: data)
+        }
+        semaphore.signal()
+    }.resume()
+    _ = semaphore.wait(timeout: DispatchTime.now() + DispatchTimeInterval.milliseconds(500))
+    return json
+}
+
+func encodeURIComponent(name: String!) -> String! {
+    return name.addingPercentEncoding(withAllowedCharacters: NSMutableCharacterSet.urlQueryAllowed)
+}
+
+class MPPlayer {
+    var displayName: String?
+
+    init(withDisplayName name: String) {
+        displayName = name;
+    }
+};
+
+class MPPlayerFriend: MPPlayer {
+    var connection: MPConnection?
+    var remotePlayerToken: String?
+    var alias: String?
+
+    convenience init(withConnection c:MPConnection,
+         remotePlayerToken r: String,
+         displayName n: String,
+         alias a: String) {
+        self.init(withDisplayName:n)
+        connection = c
+        remotePlayerToken = r
+        alias = a
+    }
+};
+
+class MPSession {
+    var connection: MPConnection?
+    var sessionToken: String?
+    var displayName: String?
+
+    init(withConnection c:MPConnection,
+         sessionToken t: String,
+         displayName n: String) {
+        connection = c
+        sessionToken = t
+        displayName = n
+    }
+};
+
+class MPLocalSession: MPSession {
+    var shareCode: String?
+    
+    convenience init(withConnection c:MPConnection,
+         localSessionToken t: String,
+         displayName n: String,
+         shareCode sc: String) {
+        self.init(withConnection: c, sessionToken: t, displayName: n)
+        shareCode = sc
+    }
+}
+
+class MPLocalPlayer: MPPlayer {
+    var connection: MPConnection?
+    var localPlayerToken: String?
+    var friendCode: String?
+    var authenticated: Bool = false
+    var sessions: [MPLocalSession] = []
+    var friends: [String] = []
+    
+    override var displayName: String? {
+        get {
+            return super.displayName
+        }
+        set(newName) {
+            guard let connection = connection, connection.isConnected else { return }
+            let comp = encodeURIComponent(name: newName!)
+            if comp == nil {
+                return
+            }
+            let url = connection.baseUrl! + "writeplayerdisplayname.json?connection=" + connection.connectionToken! + "&localPlayer=" + localPlayerToken! + "&displayName=" + comp!
+            let result = fetch(url: URL(string: url))
+            if result == nil {
+                return
+            }
+            
+            super.displayName = newName
+        }
+    }
+    
+    convenience init(withConnection c:MPConnection,
+                     localPlayerToken r: String,
+                     displayName n: String,
+                     friendCode f: String,
+                     authenticated a: Bool) {
+        self.init(withDisplayName:n)
+        connection = c
+        localPlayerToken = r
+        friendCode = f
+        authenticated = a
+    }
+
+    func httpAuthenticate() -> Bool {
```

Comment:
We should probably not use HTTP-Authentication from anything but JavaScript. Let's see how hard it would be to implement.

Reply:
https://github.com/jacques-gasselin/multiplay/projects/1#card-52056146